### PR TITLE
[CPREQ-7974] Kubernetes hardening

### PR DESCRIPTION
**Pull request by alexarefev**
### Description
* We use `kube-bench` to make `KubeMarine` CIS aligned. Some of the improvements do not supported in source code and could be applied manually.
* The suggested improvements have be reproduced

### Solution
* Create a new guide in `KubeMarine` documentations

### How to apply
Not applicable

### Checklist
- [x] I have made corresponding changes to the documentation
- [x] There is no merge conflicts

### CODE REVIEW

**Review by koryaga — APPROVED** [review]

File: documentation/internal/Hardening.md [hunk 1]
```diff
@@ -422,3 +425,55 @@ Basically, `CronJob` runs the approval command above for every CSR according to
...
+```yaml
+...
+volumeMounts:
+  - name: ingress-nginx-token
+    mountPath: /var/run/secrets/kubernetes.io/serviceaccount
+
+...
+
+volumes:
+  - name: service-account-token
```
Comment (alexarefev) [inline documentation/internal/Hardening.md:473]: The `name` doesn't match with the one in `volumeMounts`

Comment (pranavcracker) [reply documentation/internal/Hardening.md]: corrected the name.

**Comment by koryaga** [comment]
@alexarefev "are we happy" with the full content of this PR?

**Review by shmo1218 — APPROVED** [review]
The doc review is done.

File: documentation/internal/Hardening.md [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+### Auto Approval CronJob
+
+Generally, `CronJob` runs the approval command above for every CSR according to some schedule.
+
+### Auto Approval Service
+
+It is possible to install the kubelet-csr-approver service. For more information, refer to [kubelet-csr-approver](https://github.com/postfinance/kubelet-csr-approver). This service approves the CSR automatically when a CSR is created according to several settings. It is better to restrict nodes' IP addresses (`providerIpPrefixes` option) and FQDN templates (providerRegex). For more information, refer to the official documentation.
+
+## Disabling Auto-Mounting of Tokens for Service Accounts
+
```
Comment (koryaga) [inline documentation/internal/Hardening.md:429]: @pranavcracker  Pls put Kube-bench Identifier:

Comment (pranavcracker) [reply documentation/internal/Hardening.md:429]: Added Kube-bench Identifier

**Review by koryaga — APPROVED** [review]

**Pull request merged**